engine/src/main/java/org/terasology/rendering/AABBRenderer.java

```diff
@@ -90,8 +91,9 @@ public void render() {
         CoreRegistry.get(ShaderManager.class).enableDefault();
 
         glPushMatrix();
-        Vector3f cameraPosition = CoreRegistry.get(LocalPlayer.class).getViewPosition();
-        glTranslated(aabb.getCenter().x - cameraPosition.x, -cameraPosition.y, aabb.getCenter().z - cameraPosition.z);
+        Vector3f cameraPosition = CoreRegistry.get(LocalPlayer.class).getViewPosition(new Vector3f());
+        Vector3f center = aabb.center(new Vector3f());
+        glTranslated(center.x - cameraPosition.x, -cameraPosition.y, center.z - cameraPosition.z);
```

Comment:
Looking at this, I think the API for AABBs is inconsistent here. To get the min or max vector, you'll have to do separate lookups without the option to pass in a destination vector.

```
Vector3i min = new Vector3i(aabb.minX(), aabb.minY(), aabb.minZ())
```

However, for the `center` it is exactly the other way around - we can only access the center as vector, but not the individual components...

Reply:
Also, we are creating a new vector here on every render call - should we instead reuse a tmp vector instead?

Reply:
sounds like we need to modify joml-ext